### PR TITLE
Feat/new pushnotifications 01

### PR DESCRIPTION
**Pull request by cmgustavo**
* Needs (bwc v4.5.0)

https://github.com/bitpay/bitcore-wallet-client/pull/349

And

https://github.com/bitpay/bitcore-wallet-service/pull/627/files

### CODE REVIEW

**Comment by JDonadio** [comment]
Related https://github.com/bitpay/copay/issues/5001

File: src/js/services/pushNotificationsService.js [hunk 1]
```diff
@@ -1,87 +1,130 @@
...
+  root.init = function() {
+    if (!usePushNotifications || _token) return;
+    configService.whenAvailable(function(config) {
+      if (!config.pushNotificationsEnabled) return;
+    
+      $log.debug('Starting push notification registration...'); 
+
+      //Keep in mind the function will return null if the token has not been established yet.
+      FCMPlugin.getToken(function(token) {
+        $log.debug('Get token for push notifications...');
```
Comment (matiu) [inline src/js/services/pushNotificationsService.js]: Maybe we should add ` + token` to the debug message.

File: src/js/services/pushNotificationsService.js [hunk 2]
```diff
@@ -1,87 +1,130 @@
...
+    _token = null;
+  };
+
+  root.unsubscribe = function(walletClient) {
+    if (!_token) return;
+    _unsubscribe(walletClient);
+  };
+
+  var _subscribe = function(walletClient) {
+    var opts = {
```
Comment (matiu) [inline src/js/services/pushNotificationsService.js]: please add:

` if (!isIOS || !isAndroid) => exit/warn`

then:
+      platform: isIOS ? 'ios' :  'android' ,

**Comment by matiu** [comment]
Great job!

**Pull request merged**